lib/puppet/provider/exec/powershell.rb

```diff
@@ -115,6 +115,13 @@ def validatecmd(command)
     true
   end
 
+  def self.post_resource_eval
+    # TODO: make sure this isn't done in --noop and is only done when manager has been created
+    if PuppetX::PowerShell::PowerShellManager.supported?
```

Comment:
Could just be the addition of an instance variable for the provider instance indicating a manager has been created (i.e. that `run` happened)